File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ServoSteper.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.teamcode.hardware.CompBot2Hardware;

@TeleOp
public class ServoSteper extends LinearOpMode {

    CompBot2Hardware hardware;
    double position = 0.57;

    double maxServo = 1.0;
    double minServo = 0.0;

    double deltaPosition = 0;

    double deltaPosition2 = 0;

    int MotorPosition = 0;

    @Override
    public void runOpMode() throws InterruptedException {

        hardware = new CompBot2Hardware(hardwareMap);

        Servo theServo = hardware.shooterBallStop;

        waitForStart();

        while (opModeIsActive()) {


            if (gamepad1.right_bumper){
                deltaPosition += 0.01;
                if(position > maxServo){
                    position = maxServo;
                }
            }

            if (gamepad1.left_bumper){
                deltaPosition -= 0.01;
                if(position < minServo){
                    position = minServo;
                }
            }

            if (gamepad1.a){
                deltaPosition2 -= 0.01;
                if(position < minServo){
                    position = minServo;
                }
            }

            if (gamepad1.b){
                deltaPosition2 += 0.01;
                if(position < minServo){
                    position = minServo;
                }
            }
            hardware.leftKickstand.setPosition(position + deltaPosition);
            hardware.rightKickstand.setPosition(position + deltaPosition2);

//            if (gamepad1.a){
//                hardware.setIntakePower(1);
//            }
//            else {
//                hardware.setIntakePower(0);
//            }

            if (gamepad1.right_trigger > 0.5){
                MotorPosition += 10;
            }

            if (gamepad1.left_trigger > 0.5){
                MotorPosition -= 10;
            }




//            telemetry.addData("motorPosition", hardware.turret.getCurrentPosition());
            telemetry.addData("positionleft",position+deltaPosition);
            telemetry.addData("positionright",position+deltaPosition2);
            telemetry.update();

            sleep(50);
        }

    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/tools/ServoSteper.kt
package org.firstinspires.ftc.teamcode.tools

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import com.qualcomm.robotcore.hardware.Servo
import com.qualcomm.robotcore.util.ElapsedTime
import org.firstinspires.ftc.teamcode.hardware.CompBotHardware
import java.lang.Math.clamp

/**
 * steep the servo
 */
@TeleOp
class ServoSteper : LinearOpMode() {
    companion object {
        const val SPEED = 0.25 // per second
    }

    override fun runOpMode() {
        val theServo: Servo = fromHardwareCls { flipper }
//        val theServo: Servo = fromName("flipper")

        waitForStart()
        val runtime = ElapsedTime(ElapsedTime.Resolution.SECONDS)
        var lastTime = 0.0
        var pos = 0.5

        while (opModeIsActive()) {
            val now = runtime.time()
            val delta = now - lastTime
            lastTime = now

            val bumper1 = gamepad1.left_bumper
            val bumper2 = gamepad1.right_bumper
            val mod = when {
                (bumper1 && bumper2) -> 0.25
                (bumper1 || bumper2) -> 0.5
                else -> 1.0
            }
            val speed = SPEED * mod

            if (gamepad1.dpad_up) {
                pos += delta * speed
            } else if (gamepad1.dpad_down) {
                pos -= delta * speed
            }
            pos = clamp(pos, 0.0, 1.0)

            theServo.position = pos
            telemetry.addData("Position", pos)
            telemetry.addData("Speed (per s)", speed)
            telemetry.addData("Speed Mod", mod)
            telemetry.addLine("Press dpad up/down to change")
            telemetry.addLine("Press left/right bumper to slow")
            telemetry.update()
        }
    }

    private inline fun <reified T> fromName(name: String): T {
        return hardwareMap.get(T::class.java, name)
    }

    private inline fun <T> fromHardwareCls(getIt: CompBotHardware.() -> T): T {
        val instance = CompBotHardware(hardwareMap)
        return instance.getIt()
    }
}